marshal.go

```diff
@@ -0,0 +1,181 @@
+package zoekt
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"unsafe"
+)
+
+// Wire-format of map[uint32]MinimalRepoListEntry is pretty straightforward:
+//
+// byte(1) version
+// uvarint(len(minimal))
+// uvarint(sum(len(entry.Branches) for entry in minimal))
+// for repoID, entry in minimal:
+//   uvarint(repoID)
+//   byte(entry.HasSymbols)
+//   uvarint(len(entry.Branches))
+//   for b in entry.Branches:
+//     str(b.Name)
+//     str(b.Version)
+
+// reposMapEncode implements an efficient encoder for ReposMap.
+func reposMapEncode(minimal ReposMap) ([]byte, error) {
+	if minimal == nil {
+		return nil, nil
+	}
+
+	var b bytes.Buffer
+	var enc [binary.MaxVarintLen64]byte
+	varint := func(n int) {
+		m := binary.PutUvarint(enc[:], uint64(n))
+		b.Write(enc[:m])
+	}
+	str := func(s string) {
+		varint(len(s))
+		b.WriteString(s)
+	}
+	strSize := func(s string) int {
+		return binary.PutUvarint(enc[:], uint64(len(s))) + len(s)
```

Comment:
For my own understanding, we encode the length of the string as uint64 and then we add the length of the string. Why do we add the length of the string after the uint64 length of the string 🤔

Reply:
Ah wait I get it 😮‍💨 
 
[<length of string><space for the string>]